src/Raven.Server/Documents/TimeSeries/TimeSeriesStorage.cs

```diff
@@ -1052,11 +1055,197 @@ private void AddValueInternal(DateTime time, Span<double> values, Span<byte> tag
                 var inRange = timestampDiff < int.MaxValue;
                 if (inRange == false || segment.Append(_context.Allocator, (int)timestampDiff, values, tagSlice, status) == false)
                 {
+                    var segmentLastTimestamp = segment.GetLastTimestamp(BaselineDate);
+
+                    if (segmentLastTimestamp == BaselineDate)
+                    {
+                        // full segment (segment size reach to capacity/number of entries >= 65,535)
+
+                        // not from replication:
+                        // we should throw all new entries with this timestamp 
+                        if (FromReplication == false)
+                            throw new InvalidOperationException($"Segment is full. Can't add more values with {time} timestamp");
+
+                        // raise alert
+                        var msg = $"Segment reached capacity (> 2KB) and open a new segment unavailable at this point." +
+                                  $"Merge operation has performed for replication on doc: {_docId}, name: {_name}." +
+                                  $"The data after the merge may be inaccurate.";
+                        var alert = AlertRaised.Create(_context.DocumentDatabase.Name, "Time series segment is full - merge operation has performed", msg,
+                            AlertType.Replication, NotificationSeverity.Warning);
+                        _tss._documentDatabase.NotificationCenter.Add(alert);
+
+                        // merge all entries to one 
+                        if (TryMergeEntries(ref segment) == false)
+                            throw new InvalidOperationException($"Failed to merge segment values. Doc: {_docId}, name: {_name}");
+
+                        // retry append
+                        var tag = _context.GetLazyString(_tss._documentDatabase.DbBase64Id);
+                        segment.Append(_context.Allocator, (int)timestampDiff, values, SliceHolder.TagAsSpan(tag), status);
```

Comment:
Here we ran out of space - we failed in append from replication. The solution for this scenario will be to merge the entries of the segment into one,  then append again [with a new tag which is the DB-id (instead of the previous tag)]. Now we have only one entry occupied in the segment so I don't think append can fail, but I will take extra precautions and check for failure.